### DISCUSSION TITLE
編集リクエスト受諾時のContributionの高さの意図についての質問

### DISCUSSION THREAD

**Discussion by simon-ritchie** [increments/qiita-discussions#164] [ヘルプ] [answered]
興味程度となります！運営の皆様がどのような意図でそうしているのだろう・・・と気になっているため質問させていただきます！

#26 でも以下の引用のものの辺りでちらっと触れられていますが、記事投稿などがContribution1、質問回答などが0.2しかない・・・という現状で、且つ記事の多くはLGTMが0付近のものばかりの分布になっている・・・と以前の年次の集計記事などで見たことがあります！

> 1行の編集リクエストと100行の編集リクエストは同じで良いのか

記事を書いたり質問回答などはかなり手間暇かかったりする一方で↑のようなちょっとした編集リクエストでも5LGTMもらえる・・・というのは違和感が結構前から根強く感じております！（記事投稿の5倍、質問回答の25倍という差が非常に大きすぎるなと）

編集リクエストが増えた方がQiita内の記事が多く修正されて良い記事が増える・・・というのは理解できるのですが、記事投稿なども大切だと思いますのでもう少しバランスが取れている配分の方がいいのでは・・・とは感じております。

個人的には質問回答などが回答で0.2、質問者の方からLGTMが付けば合計で0.7ということを考えると編集リクエストも出した時点で0.2、受け入れられたらさらに1加えて1.2くらい・・・がバランスが良さそうくらいに思っております。

（※ゲーム業界にいるせいか、ゲームバランス的にこの辺の数字設定によるバランスなどが気になってしまい細かい点の質問となってしまい恐縮です・・・w）

この辺り運営の方々の考えが分かった方が納得感が得られる気がするのでどのような判断でこの配分になっているのかご教示いただけますと幸いです・・・！

---

※何となく現状で思いつく辺りの理由として以下のようなものを想像しております。

- 過去の集計方式を加味して、大体平均などで同じくらいになる数字で今の編集リクエストの数字を設定している（バランスというよりも過去のものと比較してユーザーに不利益が少ない形で設定する等）。
- 私も編集リクエストを受けると嬉しかったりするので、運営の皆様的にそういったポジティブな感情が生まれるアクションを増やしたいと考えていらっしゃる。
- そもそも編集リクエスト自体が総数が記事投稿などと比べると少ないのでもっと増えるように数値を設計している。
- 編集リクエストが増えることで記事の質が上がりコミュニティに取って大きくプラスになると運営の方々が考えていらっしゃる。

---

※ユーザーに対して数値が下がるマイナスのアップデートは極力入れない方が良いでしょうし、編集リクエストの既存の値を下げて欲しいというものではなく運営の皆様がどのような意図を持ってこのような設計にしているのか、どのような考えで設定されているのか・・・などに興味が沸いている形となります！

**Comment by kskwtnk** [comment, marked as answer]
@simon-ritchie 
ご意見ありがとうございます！

思いつく理由として挙げていただいた内容、どれもまさに私たちが考えていたものです！ 🙏 

特に、計算方法変更当時のユーザーランキングが大きく変わらないように調整していた点は大きいのですが、最近は若干様子も変わってきており……
ほぼ編集リクエストのみでユーザーランキング上位に入っている人もいて、simon-ritchieさんの仰るように手間と実際のContribution数に乖離が出ている感も否めず、運営でも丁度考えていたところです

回答として若干ふわっとしてしまいましたがこういった内容となっています 🙇

**Reply by simon-ritchie** [reply]
@xrxoxcxox さん

理由や経緯など理解いたしました！
ご共有ありがとうございます・・・！mm

**Reply by simon-ritchie** [reply]
@xrxoxcxox さん

https://twitter.com/Qiita/status/1506452456796221443?s=20&t=5rVdS7uZrrMv35dB8BUIog

↑とかを拝見いたしました！社内議論などありがとうございます・・・！

ゲームバランスとかでもやらかすと炎上したりする・・・のですが、（既にご存じだと思いますが）若干Twitterとか眺めていてもそういった雰囲気が出てきている感がしないでもないので色々考えることが多く大変かとは思いますが（大きな炎上などに繋がる前に）早めに手を打ったり、再集計するかどうかの判断が結構先になりそうでしたら検討中という旨のTwitter以外（トップページとかの目立つところなど）での告知を出すとか、ユーザーに対して不利益がなるべく少なくなるように編集リクエストなど大きく数字の下がりそうな方へは個別に＆早めに連絡などしておく・・・と無難かもしれません！（ランキングとかも大分荒れている印象があり、結構人の目に付いてしまうかなと）

※既にお問い合わせなどが複数の方からいっていて社内で色々対策を考えていらっしゃるとは思いますが・・・。

> QiitaのContributionを一番多く稼げる方法が「タグを1つ追加するだけの編集リクエストを送りまくる」なの、なんか納得いかない。
> https://twitter.com/YottyPG/status/1506188794222952454?s=20&t=5rVdS7uZrrMv35dB8BUIog

> Contribution稼ぐために質問に回答してるわけではないけど、調べたりソース書いたりして回答したのが、タグを1つ追加するだけの作業の25分の1のContributionって知って、なんだか悲しくなった。
> https://twitter.com/YottyPG/status/1506188795724517382?s=20&t=5rVdS7uZrrMv35dB8BUIog

> Qiitaの編集リクエスト、記事の内容の変更全くなくて、タグ付け加えるだけでしてこないでほしいな・・・Contributionを楽にあげたいのかなんなのかしらんけど。
> https://twitter.com/okaponta_/status/1505565898756673538?s=20&t=5rVdS7uZrrMv35dB8BUIog

> 自分が書いたQiitaの記事に、正直付ける意味のまったくないタグ付けの編集リクエスト届いて謎に思っていたんだけどContribution稼ぐためだったらしい。
> なるほどとなった。
> https://twitter.com/kouchi32/status/1505484623668662275?s=20&t=5rVdS7uZrrMv35dB8BUIog

> Qiitaで凄いcontributionあげてるからどんな記事かなと探したけどさほどLGTMは付いておらず。履歴を確認したら編集リクエストが沢山出ていてレビュー力高いのかと編集履歴を見たら全部がタグの追加でした。
ちょいと真似てしまったがこれは違うと思って思いとどまりました。
> https://twitter.com/daikin_project/status/1503227720570122242?s=20&t=5rVdS7uZrrMv35dB8BUIog

> ちょっと前に Qiita で書いた記事に、タグを追加するだけという編集リクエストが。
そしてそのタグ、記事０件。ド新規。今まで誰も使ったことがない。ナンダコレ。
> https://twitter.com/licht_dol/status/1496877389812408324?s=20&t=5rVdS7uZrrMv35dB8BUIog

※ランキングのための極端なことをされている方も学生でQiita Organizationで所属が出てしまっている・・・というのも余計に今回のケースだと良くないかなと・・・！（今後さらに批判が増えて来て収拾がつかなくなってくると、学生でその辺の行為が悪感情を発生させてしまうというのが分からずにやっている＆大学側もそういった意図は無かった＆Qiita側にもお問い合わせが増えたり悪評が出たりする＆他のユーザーからしても不快になったり・・・とこのまま炎上していったら誰も得をしない感じになるかなと）

なんだか昔も似たようなことがあって別の大学名が出ていて、燃えてコメントやSNSで大学側へ悪評が・・・みたいなことがあったような気がしますが、大学側に取っても良くないかなと・・・！

**Reply by okaponta** [reply]
@simon-ritchie 
ちょうど自分も同じ方から3回目のタグだけ追加のリクエストきて、Discussionあげようとしていました！
Discussion投稿ありがとうございます 🙇

**Reply by kskwtnk** [reply]
@simon-ritchie 
追加でのご意見ありがとうございます！
しかもかなり詳細に……Twitterでのみなさんの意見まで拾ってきていただいて大変ありがたいです 🙏 

議論中の内容をもっとこまめに分かりやすく共有すべきだったと反省している最中ですが、たった今この件についてブログを出しました 🙇 

https://blog.qiita.com/change-contribution-calculation-20220401/

色々とご心配をおかけして申し訳ありませんが、今後ともよろしくお願いいたします 🙏

**Reply by simon-ritchie** [reply]
@okaponta さん

それは良かったです！

**Reply by simon-ritchie** [reply]
@xrxoxcxox さん

おお・・・承知いたしました！
（この辺り短期間での決定は中々骨が折れると思いますが）非常に迅速なご対応ありがとうございます・・・！🙇‍♂️
（なんだかプレッシャーを与えてしまった感があり恐縮です・・・）

ブログも拝見いたました！ご共有ありがとうございます！

**Reply by flowercaean** [reply]
> @simon-ritchie ちょうど自分も同じ方から3回目のタグだけ追加のリクエストきて、Discussionあげようとしていました！ Discussion投稿ありがとうございます 🙇

私も以前そういうことがあったので、その時使った対処法を....
そういった場合は、ミュート機能を活用するといいかもです。
こんな感じで、ミュートしたいユーザーのページから、...を押して、ミュートすれば、通知が来なくなって、快適です。
![image](https://user-images.githubusercontent.com/102569114/160524659-1226653e-ff06-4eb6-bcb7-f190fd227e75.png)

**Reply by simon-ritchie** [reply]
@flowercaean さん

ご共有ありがとうございます！

**Comment by JunichiIto** [comment]
気になったのでどのアカウントがどんな編集リクエストを送っているのか調べてみましたが、件のアカウントは手作業ではなく「記事タイトルから適当なキーワードを抜き出して編集リクエストを送りつけるスクリプト」を作ってそうな印象ですね。。

こういう使い方は利用規約的にBANできないのかな、とか思ったりしました（該当する利用規約があるかどうかはちゃんと調べていませんが）。

**Reply by simon-ritchie** [reply]
@JunichiIto さん

おお、左様ですね・・・（タグ詳細とかまでは追っておりませんでした）
ご共有ありがとうございます！

今軽く利用規約再読してみていましたがぴったり「これ！」という項目は無さそうで、もしかしたら該当するかもくらいのレベルでも以下の辺とかなのかな・・・？という感じがございます。

> 社会一般的観点より倫理的に問題があると認められる行為、第三者に不快感を与える内容の情報を開示する行為。
> ...
> 迷惑行為、嫌がらせ行為、誹謗中傷行為、正当な権利なく第三者に精神的被害・経済的被害を与える行為

もしスクリプトの場合Contributionの数値に調整が入っても止まらないかもしれませんね・・・。  
（今日のユーザーランキング見ていたら告知が出た後にむしろ編集リクエストなど送るペースが加速している感が・・・w）

後は膨大に編集リクエストを送られて、メールなどがユーザーにたくさん飛んでそれを見たユーザーによるアクセス集中などでサーバーに負担をかけている・・・といったところは何らか抵触するかもしれませんが・・・（その辺のサーバーに高負荷をかけたりなどは規約上には無さそうな感じも・・・）。

---

@xrxoxcxox さん

この辺はユーザー側からはログを見たり判断したりはできるものでもありませんので、お手数ではございますが @JunichiIto さんのコメントをご確認いただけますと幸いです・・・！🙇‍♂️

（スクリプトとかによる実行かどうかはログをご確認いただいて深夜帯とかにも編集リクエスト送っていたり一日中送っていたり・・・とかも判断の基準になるかもしれません）

**Reply by simon-ritchie** [reply]
> こういう使い方は利用規約的にBANできないのかな、とか思ったりしました（該当する利用規約があるかどうかはちゃんと調べていませんが）。

直近の皆さんのTwitterの反応見ていた感じ、BANとかは色々な事情で難しくとも件の方だけは再集計・ランキング除外・・・とかは入った方が無難なのかもしれませんが・・・。

この辺は正解が難しそうです・・・💦
（少なくとも判断によってはユーザーに不利益が出うるので結構慎重に判断いただく必要がありそうではありますが・・・一方でこの手のものを「やったもん勝ち」な状態が良くないというのも分かるので難しそうな印象が・・・）

> Qiitaの編集リクエストするだけでContributionsが増えるの納得できない。リクエスト送るだけで増えるのは貢献しているとは言い難い気がする。
> https://twitter.com/_EorF/status/1508746144733548545?s=20&t=tE9dTXUeR28FRWxoigLG1g

> ただタグ追加リクエスト送ってるだけ表彰プログラムに受賞するのはどうなのか。到底エンジニアコミュニティに貢献してるとは思えない。
> https://twitter.com/_EorF/status/1508747917762654219?s=20&t=tE9dTXUeR28FRWxoigLG1g

> 変更されるのはありがたいけど、再計算はされないのかー…。
> https://twitter.com/YottyPG/status/1508668658440347650?s=20&t=tE9dTXUeR28FRWxoigLG1g

> この人記事に比べてやけにcontribution多いと思ったら、タグ追加のみの編集リクエストをひたすら出してcontribution稼いでいるのか。evilじゃん。
> https://twitter.com/zhongdaojianta5/status/1508660292552589315?s=20&t=tE9dTXUeR28FRWxoigLG1g

> この変更だとコミュニティに貢献しない「Contribution稼ぎのためのタグ付けだけの編集リクエスト」の根本的解決になっていないので、編集リクエストの最低文字数を規定して欲しかった
> https://twitter.com/c60evaporator/status/1508608940581748737?s=20&t=tE9dTXUeR28FRWxoigLG1g

**Reply by kskwtnk** [reply]
@JunichiIto @simon-ritchie 
おふたりとも、詳細まで確認いただいてありがとうございます！

送信頻度なども確認してスクリプトの線はあるのですが、いかんせんそれを取り締まる規約がなく……我々の落ち度です
simon-ritchieさんが出してくださった規約も、明確に誰かの被害になっているわけではない（感情的には、というのはあるのですが）ので規約違反とは言えない……という判断です

規約内容に反しないアクションについて、いきなり我々から直接的な介入をするというのも、これまでのQiitaの色々を見ていると実施しづらく……

ただ、仰るようにやったもん勝ちになってしまうのは避けたいため、少なくとも今後は同じ手は使えないように調整しようと思っています
現段階で公表できることが少なく申し訳ありません 🙇

**Reply by JunichiIto** [reply]
@xrxoxcxox なるほど、了解しました。バランスの取り方が難しいところですね。いろいろ大変だと思いますが、引き続きご対応よろしくお願いします！

**Reply by simon-ritchie** [reply]
@xrxoxcxox さん

委細承知いたしました！ご共有ありがとうございます・・・！🙇

**Reply by shiracamus** [reply]
送信頻度による制限機能が実装されたようですね。
それ自体はいいことだと思うのですが、8時間くらい前に編集リクエストした後ですが制限がかかりました。
どれくらいの間隔や頻度・回数で制限がかかるのでしょうか？
もしかすると、制限中に再度編集リクエストすると制限が延びる仕組みでしょうか？
![image](https://user-images.githubusercontent.com/4151943/160854503-bc089826-2e4e-46f6-8293-7fb3b4bba142.png)

**Reply by kskwtnk** [reply]
@shiracamus 
すみません！こちら、詳細な頻度や回数は明かさない方向で話をしています……ご容赦ください 🙏
（○時間で○回までなら、の前提でスクリプトなどが組まれてもまた難しいことになってしまうため……）

ただ、導入直後なのもあり、実際のみなさんの状況を見て基準やロジックは変えていく予定です
 shiracamusさんはかなり多く編集リクエストを送ってくださっているのでご迷惑をおかけしてしまうかもしれません
何卒ご理解いただけますと幸いです

**Reply by shiracamus** [reply]
条件を明かさない件は承知しました。スパム対策と同様ですね。
先ほどいくつか編集リクエストを出してみましたが制限はかかりませんでした。
これくらいの頻度で制限かからなければ大丈夫そうです。
